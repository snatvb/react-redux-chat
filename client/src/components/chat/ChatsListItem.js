/**
 * Created by snatvb on 17.12.16.
 */

import React, { Component } from 'react';
import { browserHistory } from 'react-router';
//import {connect} from 'react-redux';
//import * as chatUserActions from '../../actions/chatUser';

class ChatListItem extends Component {
    constructor() {
        super();

        this.handlerClick = this.handlerClick.bind(this);
    }

    handlerClick () {
        const {chat} = this.props;
        //this.props.dispatch(chatUserActions.setChatUser(user));
        browserHistory.push(`/chat/${chat.id}`);
    }

    render() {
        const {chat} = this.props;
        const classNameContainer = this.props.selected ? 'chat-list_item selected' : 'chat-list_item';
        return (
            <div className={classNameContainer} onClick={this.handlerClick}>
                {chat.user.name}
                {chat.user.online && <div className="chat-list_item_online">Онлайн</div>}
            </div>
        );
    }
}

ChatListItem.propTypes = {
    chat: React.PropTypes.shape({
        id: React.PropTypes.number.isRequired,
        user: React.PropTypes.shape({
            id: React.PropTypes.number.isRequired,
            name: React.PropTypes.string.isRequired
        }).isRequired
    }).isRequired
};

export default ChatListItem;