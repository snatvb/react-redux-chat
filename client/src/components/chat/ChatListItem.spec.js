/**
 * Created by snatvb on 19.12.16.
 */

import React from 'react';
import ChatsListItem from './ChatsListItem';
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer';


describe('Chats List Item component', () => {
    it('render list chats', () => {
        const props = {
            chat: {
                id: 0,
                user: {
                    id: 23432,
                    name: "tester"
                }
            }
        };
        const component = renderer.create(
            <ChatsListItem {...props} />
        );
        expect(component).toMatchSnapshot();
    });
    it('render list chats when online', () => {
        const props = {
            chat: {
                id: 1,
                user: {
                    id: 234324,
                    name: "tester 2",
                    online: true
                }
            }
        };
        const component = renderer.create(
            <ChatsListItem {...props} />
        );
        expect(component).toMatchSnapshot();
    });
});