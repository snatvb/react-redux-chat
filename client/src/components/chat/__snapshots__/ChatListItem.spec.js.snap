exports[`Chats List Item component render list chats 1`] = `
<div
  className="chat-list_item"
  onClick={[Function]}>
  tester
</div>
`;

exports[`Chats List Item component render list chats when online 1`] = `
<div
  className="chat-list_item"
  onClick={[Function]}>
  tester 2
  <div
    className="chat-list_item_online">
    Онлайн
  </div>
</div>
`;
